packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -165,6 +180,27 @@ export function getChildHostContext(
   return getChildNamespace(parentNamespace, type);
 }
 
+export function getChildHostContextForEvent(
+  parentHostContext: HostContext,
+  event: ReactEventComponent | ReactEventTarget,
+): HostContext {
+  if (__DEV__) {
+    const parentHostContextDev = ((parentHostContext: any): HostContextDev);
+    if (event.$$typeof === REACT_EVENT_COMPONENT_TYPE) {
```

Comment:
It might be nicer to pass a flag or the symbol as an argument and whatever else you need as an argument. That way the various host configs aren't so tied to the implementation details of the VDOM and the structure of this object. E.g. for createInstance we pass `type` and `props` as explicit arguments instead of passing the whole React Element.